docs/docs/tutorial/chapter3/saving-data.md

```diff
@@ -297,35 +440,39 @@ const ContactPage = () => {
 export default ContactPage
 ```
 
-We reference the `createContact` mutation we defined in the Contacts SDL passing it an `input` object which will contain the actual name, email and message values.
-
-Next we'll call the `useMutation` hook provided by Redwood which will allow us to execute the mutation when we're ready (don't forget to `import` it):
+</TabItem>
+<TabItem value="ts" label="TypeScript">
 
-```jsx title="web/src/pages/ContactPage/ContactPage.js"
-// highlight-next-line
-import { MetaTags, useMutation } from '@redwoodjs/web'
+```tsx title="web/src/pages/ContactPage/ContactPage.tsx"
+import { MetaTags } from '@redwoodjs/web'
 import {
   FieldError,
   Form,
   Label,
   TextField,
   TextAreaField,
   Submit,
+  SubmitHandler,
 } from '@redwoodjs/forms'
 
+// highlight-start
```

Comment:
```suggestion
// highlight-start
import { CreateContactMutation, CreateContactInput } from 'types/graphql'
```

We should show using these types

Reply:
Thanks for pointing this out. I didn't do it exactly like you've suggested, but I do use generated types more now, and I'm really happy with how it turned out. Using `CreateContactMutationVariables` together with my custom `FormValues` gets me exactly the kind of type errors I'm looking for if something doesn't match up.